packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -82,20 +100,66 @@ class Carousel extends Foundation<
      */
     public render(): React.ReactElement<HTMLDivElement> {
         return (
-            <div {...this.unhandledProps()} className={this.generateClassNames()}>
+            <div
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                ref={this.rootEl}
+                onFocus={this.handleCarouselHoverOrFocus}
+                onMouseEnter={this.handleCarouselHoverOrFocus}
+                onMouseLeave={this.handleCarouselMouseLeave}
+            >
                 {this.generatePreviousFlipper()}
                 <Tabs
                     label={this.props.label}
                     activeId={this.state.activeId}
                     onUpdate={this.handleUpdate}
                     items={this.slides as TabsItem[]}
                     managedClasses={this.generateTabsClassNames()}
+                    disableTabFocus={this.props.autoplay}
                 />
                 {this.generateNextFlipper()}
             </div>
         );
     }
 
+    /**
+     * React lifecycle hook
+     */
+    public componentDidMount(): void {
+        if (canUseDOM() && this.props.autoplay) {
+            // Set initial interval for autoplay
+            this.autoplayTimer = window.setInterval(
+                this.nextSlide,
+                this.autoplayInterval
+            );
+        }
+    }
+
+    /**
+     * React lifecycle hook
+     */
+    public componentDidUpdate(prevProps: CarouselProps, prevState: CarouselState): void {
+        if (this.props.autoplay && isNil(this.autoplayTimer)) {
+            // Set the window interval if we are in autplay and don't have a timer
+            this.autoplayTimer = window.setInterval(
+                this.nextSlide,
+                this.autoplayInterval
+            );
+        } else if (!this.props.autoplay && !isNil(this.autoplayTimer)) {
+            // Clear the interval if we should not be autoplaying
+            this.autoplayTimer = window.clearInterval(this.autoplayTimer as number);
+        }
+    }
+
+    /**
+     * React lifecycle hook
+     */
+    // public componentDidUnmount(): void {
+    //     if (!isNil(this.autoplayTimer)) {
+    //         this.autoplayTimer = window.clearInterval(this.autoplayTimer as number);
+    //     }
+    // }
```

Comment:
still needed?

Reply:
thought I pushed this. I think it needs to be there, but shouldn’t be uncommented.